product_pack/models/product_product.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api
+import math
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    pack_line_ids = fields.One2many(
+        'product.pack.line',
+        'parent_product_id',
+        'Pack Products',
+        help='Products that are part of this pack.'
+    )
+    used_pack_line_ids = fields.One2many(
+        'product.pack.line',
+        'product_id',
+        'Found in packs',
+        help='Packs where product is used.'
+    )
+
+    @api.multi
+    def split_pack_products(self):
+        """Split products and the pack in 2 separate recordsets.
+        :return: [packs, no_packs]
+        """
+        packs = self.filtered(lambda p: p.pack and p.pack_type in {
+            'totaliced_price',
+            'none_detailed_assisted_price',
+            'none_detailed_totaliced_price',
+        })
+        # TODO: Check why this is needed
+        # for compatibility with website_sale
+        if self._context.get('website_id', False) and \
+                not self._context.get('from_cart', False):
+            packs |= self.filtered(
+                lambda p: p.pack and p.pack_type == 'components_price')
+
+        no_packs = (self | self.mapped('pack_line_ids.product_id')) - packs
+        return packs, no_packs
+
+    @api.multi
+    def price_compute(self, price_type, uom=False, currency=False,
+                      company=False):
+        packs, no_packs = self.split_pack_products()
+        prices = super(ProductProduct, no_packs).price_compute(
+            price_type, uom, currency, company)
+        for product in packs.with_context(prefetch_fields=False):
+            pack_price = 0.0
+            for pack_line in product.pack_line_ids:
+                product_line_price = pack_line.product_id.price
+                pack_price += (product_line_price * pack_line.quantity)
+            prices[product.id] = pack_price
+        return prices
+
+    @api.depends('list_price', 'price_extra')
+    def _compute_product_lst_price(self):
+        packs, no_packs = self.split_pack_products()
+        super(ProductProduct, no_packs)._compute_product_lst_price()
+
+        to_uom = None
+        if 'uom' in self._context:
+            to_uom = self.env['uom.uom'].browse([self._context['uom']])
+
+        for product in packs:
+            list_price = product.price_compute('list_price').get(product.id)
+            if to_uom:
+                list_price = product.uom_id._compute_price(
+                    list_price, to_uom)
+            product.lst_price = list_price + product.price_extra
+
+    def _compute_quantities_dict(
```

Comment:
This method is defined in stock module (https://github.com/odoo/odoo/blob/12.0/addons/stock/models/product.py#L92), so I think that mybe you should create a new module for extend this or add stock dependency 
cc @pedrobaeza What do you think about this??

Reply:
Maybe add a module named stock_product_pack or something like that, which depends on product_pack and stock and with autoinstall true. The only rare thing I see is that this module will only have that method

Reply:
Is this related to only sale? If so, there should be a module `sale_stock_product_pack` (as we can sell services pack and don't need `stock` for that). If it can be in both sale and purchase, then we need a module `stock_product_pack`, and probably their corresponding glue modules.